src/main/java/org/springframework/integration/aws/config/xml/SqsOutboundChannelAdapterParser.java

```diff
@@ -34,13 +34,37 @@
  */
 public class SqsOutboundChannelAdapterParser extends AbstractOutboundChannelAdapterParser {
 
+    public static final String QUEUE_MESSAGING_TEMPLATE_REF = "queue-messaging-template";
+
+    private boolean hasAttribute(Element element, String attribute) {
+        return StringUtils.hasText(element.getAttribute(attribute));
+    }
+
 	@Override
 	protected AbstractBeanDefinition parseConsumer(Element element, ParserContext parserContext) {
 		BeanDefinitionBuilder builder = BeanDefinitionBuilder.genericBeanDefinition(SqsMessageHandler.class);
-		builder.addConstructorArgReference(element.getAttribute(AmazonWSParserUtils.SQS_REF));
-		if (StringUtils.hasText(element.getAttribute(AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF))) {
+
+		final boolean hasQueueMessagingTemplate = hasAttribute(element, QUEUE_MESSAGING_TEMPLATE_REF);
+		final boolean hasSqs = hasAttribute(element, AmazonWSParserUtils.SQS_REF);
+		final boolean hasResourceResolver = hasAttribute(element, AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF);
+		if (hasQueueMessagingTemplate && (hasSqs || hasResourceResolver)) {
+			parserContext.getReaderContext().error(QUEUE_MESSAGING_TEMPLATE_REF +
+					" should not be defined in conjunction with " + AmazonWSParserUtils.SQS_REF + " or " + AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF, element);
+		}
+
+        if (!hasQueueMessagingTemplate && !hasSqs) {
```

Comment:
I meant **all** indentations vulnerabilities :smile: . See changes for the `SqsMessageHandler` from this your PR: https://github.com/spring-projects/spring-integration-aws/pull/8/files#diff-9306d23280a85feb98b4d91b4d101140R53 
Please, find here some IDEA option for Spring code style: https://github.com/spring-projects/spring-framework/blob/master/src/idea/spring-framework.xml 
Plus these rules: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Code-Style